### PR TITLE
[FLINK-12746][docs][flip-42] Getting Started - Project Setup and DataStream Example Walkthrough

### PR DESCRIPTION
**Pull request by sjwiesman**
## What is the purpose of the change
Move project setup to the new getting started guide and remove the outdated examples section from the documentation.

## Brief change log

See commit messages.

### CODE REVIEW

**Comment by sjwiesman** [comment]
cc @knaufk

**Comment by knaufk** [comment]
Thanks for starting this @sjwiesman. I think, we need to do more than moving the current files: 
* TableAPI and DataStream API will be the two primary APIs going forward. So, I think, we should expand the quickstarts to also contain the TableAPI for Java and Scala. We will keep the DataSet API around until it is officially deprecated.
*  For completeness, we should add a Python Table API Project Setup section, but this can be follow up ticket. Do you create one?

File: docs/getting-started/projectsetup/java_api_quickstart.md [hunk 1]
```diff
@@ -23,19 +23,16 @@ specific language governing permissions and limitations
...
-* This will be replaced by the TOC
-{:toc}
-
-
-## Build Tools
-
-Flink projects can be built with different build tools.
-In order to get started quickly, Flink provides project templates for the following build tools:
+Flink projects can be built with a number of different build tools.
+In order to get started quickly, the community offer project templates for the following build tools:
```
Comment (knaufk) [inline docs/getting-started/projectsetup/java_api_quickstart.md]: offers

File: docs/getting-started/projectsetup/java_api_quickstart.md [hunk 2]
```diff
@@ -101,7 +98,7 @@ quickstart/
             └── log4j.properties
 {% endhighlight %}
 
-The sample project is a __Maven project__, which contains two classes: _StreamingJob_ and _BatchJob_ are the basic skeleton programs for a *DataStream* and *DataSet* program.
+The sample project is a __Maven project__, which contains two classes: _StreamingJob_ and _BatchJob_ are the basic skeleton programs for *DataStream* and *DataSet* applications.
```
Comment (knaufk) [inline docs/getting-started/projectsetup/java_api_quickstart.md]: Add TableAPI

File: docs/getting-started/projectsetup/java_api_quickstart.md [hunk 3]
```diff
@@ -336,10 +333,7 @@ can run the application from the JAR file without additionally specifying the ma
 Write your application!
 
 If you are writing a streaming application and you are looking for inspiration what to write,
-take a look at the [Stream Processing Application Tutorial]({{ site.baseurl }}/tutorials/datastream_api.html#writing-a-flink-program).
-
-If you are writing a batch processing application and you are looking for inspiration what to write,
-take a look at the [Batch Application Examples]({{ site.baseurl }}/dev/batch/examples.html).
+take a look at the [Stream Processing Application Tutorial]({{ site.baseurl }}/getting-started/tutorials/datastream_api.html#writing-a-flink-program).
```
Comment (knaufk) [inline docs/getting-started/projectsetup/java_api_quickstart.md]: We will remove the Tutorials as part of this FLIP. So in the future these links should probably only point examples and code walk throughs. What do you think?

File: docs/getting-started/projectsetup/scala_api_quickstart.md [hunk 4]
```diff
@@ -23,19 +23,17 @@ specific language governing permissions and limitations
...
-* This will be replaced by the TOC
-{:toc}
-
-
-## Build Tools
 
-Flink projects can be built with different build tools.
-In order to get started quickly, Flink provides project templates for the following build tools:
+Flink projects can be built with a number of different build tools.
+In order to get started quickly, the community offer project templates for the following build tools:
```
Comment (knaufk) [inline docs/getting-started/projectsetup/scala_api_quickstart.md]: offers

File: docs/getting-started/projectsetup/scala_api_quickstart.md [hunk 5]
```diff
@@ -212,12 +210,9 @@ can run time application from the JAR file without additionally specifying the m
 Write your application!
 
 If you are writing a streaming application and you are looking for inspiration what to write,
-take a look at the [Stream Processing Application Tutorial]({{ site.baseurl }}/tutorials/datastream_api.html#writing-a-flink-program)
-
-If you are writing a batch processing application and you are looking for inspiration what to write,
-take a look at the [Batch Application Examples]({{ site.baseurl }}/dev/batch/examples.html)
+take a look at the [Stream Processing Application Tutorial]({{ site.baseurl }}/getting-started/tutorials/datastream_api.html#writing-a-flink-program).
```
Comment (knaufk) [inline docs/getting-started/projectsetup/scala_api_quickstart.md]: see above

File: docs/getting-started/projectsetup/java_api_quickstart.md [hunk 6]
```diff
@@ -336,10 +333,7 @@ can run the application from the JAR file without additionally specifying the ma
...
 
 For a complete overview over the APIs, have a look at the
 [DataStream API]({{ site.baseurl }}/dev/datastream_api.html) and
```
Comment (knaufk) [inline docs/getting-started/projectsetup/java_api_quickstart.md]: add Table API

File: docs/getting-started/projectsetup/scala_api_quickstart.md [hunk 7]
```diff
@@ -212,12 +210,9 @@ can run time application from the JAR file without additionally specifying the m
...
 
-For a complete overview over the APIa, have a look at the
+For a complete overview over the APIs, have a look at the
 [DataStream API]({{ site.baseurl }}/dev/datastream_api.html) and
 [DataSet API]({{ site.baseurl }}/dev/batch/index.html) sections.
```
Comment (knaufk) [inline docs/getting-started/projectsetup/scala_api_quickstart.md]: add TableAPI

**Comment by sjwiesman** [comment]
Closing, I'm going to break this up into some sub-components.

**Pull request closed without merging**